File: BackEnd/src/main/java/com/Checkerly/BackEnd/dto/UserDTO.java
package com.Checkerly.BackEnd.dto;

import java.io.Serial;
import java.io.Serializable;

import com.Checkerly.BackEnd.domain.User;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

public class UserDTO implements Serializable{
    @Serial
	private static final long serialVersionUID = 1L;
	
	private String id;
	private String name;
	private String senha;
	private String email;
	private String celular;
	
	public UserDTO(User obj) {
		id = obj.getId();
		name = obj.getName();
		senha = obj.getSenha();
		email = obj.getEmail();
		celular = obj.getCelular();
	}
}

File: BackEnd/src/main/kotlin/com/Checkerly/BackEnd/dto/UserDTO.kt
package com.Checkerly.BackEnd.dto

import com.Checkerly.BackEnd.domain.User
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.io.Serial
import java.io.Serializable

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
class UserDTO(obj: User) : Serializable {
    private val id: String?
    private val name: String?
    private val senha: String?
    private val email: String?
    private val celular: String?

    init {
        id = obj.getId()
        name = obj.getName()
        senha = obj.getSenha()
        email = obj.getEmail()
        celular = obj.getCelular()
    }

    companion object {
        @Serial
        private const val serialVersionUID: Long = 1L
    }
}